### PR TITLE
adding odgi/stats module

### PR DESCRIPTION
**Pull request by subwaystation**
## PR checklist

Contributes to https://github.com/nf-core/pangenome/issues/70.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

This PR adds the odgi/stats module which is required for the https://github.com/nf-core/pangenome pipeline.

### CODE REVIEW

**Comment by subwaystation** [comment]
@heuermh @ggabernet Could you please take a look at this?

The nf-core linting fails, because the md5sum is of an empty file. Which in this case is true, because running `odgi stats` without any arguments emits only an empty file.

However, I am puzzled about the failing workflows. What's the problem here? Thanks for any help!

**Comment by ggabernet** [comment]
It seems like your `test.yml` does not follow the specifications. How did you create it? Can you try re-creating it with the `nf-core modules create-test-yml` command?

**Comment by subwaystation** [comment]
It was automatically created when I did `nf-core modules create odgi/stats` following https://nf-co.re/developers/tutorials/dsl2_modules_tutorial#create-the-module-template.

**Comment by subwaystation** [comment]
How shall I fix the md5sum problem @ggabernet? I can add an argument to this test case, so it does not produce an empty file. Would that be fine? Or is there a way to ignore it?

**Comment by subwaystation** [comment]
As suggested by @mirpedrol I just don't check for the md5sum and only for file existence.

File: modules/nf-core/odgi/stats/main.nf [hunk 1]
```diff
@@ -0,0 +1,36 @@
+process ODGI_STATS {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::odgi=0.8.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/odgi:0.8.0--py39h2add14b_0':
+        'quay.io/biocontainers/odgi:0.8.0--py310hc8f18ef_0' }"
```
Comment (mirpedrol) [inline modules/nf-core/odgi/stats/main.nf:8]: The container version is different, is it possible to provide the same?

File: modules/nf-core/odgi/stats/main.nf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    input:
+    tuple val(meta), path(graph)
+    val(suffix)
```
Comment (mirpedrol) [inline modules/nf-core/odgi/stats/main.nf:12]: Should be passed through `$args` (see https://nf-co.re/docs/contributing/modules#general) or maybe you can obtain it from the parameters in `$args` if I understand the documentation right

**Review by mirpedrol — COMMENTED** [review]
Looks good! I only added a couple of comments :)

File: modules/nf-core/odgi/stats/main.nf (hunk 2 above)
Comment (subwaystation) [reply modules/nf-core/odgi/stats/main.nf:12]: I thought the module itself should be able to run through without any optional arguments from `$args`. So the argument is mandatory in the sense that the module just won't run properly. But let me change.

File: modules/nf-core/odgi/stats/main.nf (hunk 1 above)
Comment (subwaystation) [reply modules/nf-core/odgi/stats/main.nf:8]: There is no py310 version on singularity available, so let me downgrade.

File: modules/nf-core/odgi/stats/main.nf (hunk 2 above)
Comment (subwaystation) [reply modules/nf-core/odgi/stats/main.nf:12]: @ggabernet I discussed a lot with @mirpedrol and the usual way would be to do something like:
`def suffix = args.contains("--tsv") ? ".tsv" : ".yaml"`. However, I already need the `suffix` when describing the outputs of the module. So using `$args` is not an option. Can I still specify an additional mandatory input channel as shown in the tutorial https://nf-co.re/developers/tutorials/dsl2_modules_tutorial#create-a-test-workflow?

What other solutions would you suggest? Thanks for any help!

Comment (subwaystation) [reply modules/nf-core/odgi/stats/main.nf]: As shown by @mirpedrol I am following her example https://github.com/nf-core/modules/pull/1874#pullrequestreview-1147515478 and use optional outputs. This is a straightforward solution and eliminates the requirement of an extra input channel.

**Review by mirpedrol — APPROVED** [review]
LGTM 🚀
